### PR TITLE
Add 'details' command

### PR DESCRIPTION
**Pull request by l33loo**
The command returns the Google Maps details of a place if there is a single result, or a list of the max top 10 results.

Includes:
- Refactoring Google Maps API piece into separate file and class
- Updating Sorbet
- Updating README
- Adding tests and related VCR cassettes

Note: the 'google-maps' gem doesn't expose Google Maps' 'Search Nearby' functionality, only the 'Details' one, so the 'details' command here is a bit rigid as it is not possible to search for a type of place near a specific location (e.g., restaurants near Seattle WA).  I might eventually look into other gems to see if we can find one that exposes 'Search Nearby'.

### CODE REVIEW

**Comment by dbalatero** [comment]
Ah sorry! Been moving all month and missed this. I will look today.

**Comment by l33loo** [comment]
No worries, I figured you were busy :)

File: README.md [hunk 1]
```diff
@@ -73,6 +74,49 @@ this service!
 
 Here are the currently supported commands. If you want to add more commands, see [Contributing a new command](#contributing-a-new-command).
 
+### details
+
+This command will text you the available info about a specific place listed on Google Maps.
```
Comment (dbalatero) [inline README.md]: Thanks for the great docs!

File: lib/commands/details.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        details living computers museum, seattle wa
+        details korean restaurant victoria bc
+      HELP
+    end
+
+    sig { override.returns(String) }
+    def response_body
+      return api_key_missing unless api_key_exists?
+
+      return help if arg_text === ''
```
Comment (dbalatero) [inline lib/commands/details.rb]: I'd use `==` double equals here, as triple equals semantics are a bit weird: https://medium.com/rubyinside/triple-equals-black-magic-d934936a6379

OR you can do `if arg_text.empty?` if that reads prettier to you

File: lib/commands/details.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+      begin
+        places = Google::Maps.places(arg_text)
+        place = Google::Maps.place(places.first.place_id)
+      rescue StandardError => error
+        return "NO RESULTS FOR \"#{arg_text}\"" if error.message === 'Google did not return any results: ZERO_RESULTS'
+        return error.message
+      rescue
```
Comment (dbalatero) [inline lib/commands/details.rb]: If you give a `rescue` block with no error type, it defaults to `StandardError` anyways, so this block is redundant. If you're trying to capture every error ever, you need to `rescue Exception`, which is a parent class of `StandardError`.

File: lib/commands/details.rb [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    sig { override.returns(String) }
+    def response_body
+      return api_key_missing unless api_key_exists?
+
+      return help if arg_text === ''
+
+      begin
+        places = Google::Maps.places(arg_text)
+        place = Google::Maps.place(places.first.place_id)
+      rescue StandardError => error
```
Comment (dbalatero) [inline lib/commands/details.rb]: Is there a more specific error you could capture? What error does `places()` or `place()` actually raise?

File: lib/commands/details.rb [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        return "NO RESULTS FOR \"#{arg_text}\"" if error.message === 'Google did not return any results: ZERO_RESULTS'
+        return error.message
+      rescue
+        return 'THERE WAS AN ERROR'
+      end
+
+      return "#{places.slice(0, 10).map.with_index { |e, i| "(#{i + 1}) #{e}" }.join(%Q{\n})}" if places.length > 1
+
+      place_name = place.name
```
Comment (dbalatero) [inline lib/commands/details.rb]: You could pull line 45-59 out to:

```ruby
private

sig {params(place: Google::Maps::Place)} # not sure correct type
def build_response(place)
  # ...

  "#{place_name}\n#{place_address}#{data_phone}#{data_rating}"\
        "#{data_price_level}#{data_permanently_closed}#{open_now}#{weekday_hours}"
end
```

File: lib/commands/details.rb [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        places = Google::Maps.places(arg_text)
+        place = Google::Maps.place(places.first.place_id)
+      rescue StandardError => error
+        return "NO RESULTS FOR \"#{arg_text}\"" if error.message === 'Google did not return any results: ZERO_RESULTS'
+        return error.message
+      rescue
+        return 'THERE WAS AN ERROR'
+      end
+
+      return "#{places.slice(0, 10).map.with_index { |e, i| "(#{i + 1}) #{e}" }.join(%Q{\n})}" if places.length > 1
```
Comment (dbalatero) [inline lib/commands/details.rb]: You can optionally pull this out into its own method with a specific name:

```ruby
private

sig { params(places: T::Array[Google::Maps::Place]) }
def list_of_places(places)
  # use `place` as the var name, `e` is not as clear to me
  places.slice(0, 10).map_with_index do |place, i|
    # ...
  end
end
```

and call it
```ruby
return list_of_places(places) if places.length > 1
```

Comment (dbalatero) [reply lib/commands/details.rb]: I like pulling out chunks of functionality into well-named methods, as you can start reading this method down in more of an "English", conversational kind of way - vs. having to parse what the raw`map_with_index` statement is doing.

**Comment by l33loo** [comment]
Thanks @dbalatero for another awesome code review!

FYI, I did not regenerate the rbi files this time around -- it didn't seem relevant but I could be wrong!

**Comment by dbalatero** [comment]
@l33loo Looks great, thank you for addressing the feedback!

**Pull request merged**